DCiA/dcia/construct_graph.py

```diff
@@ -0,0 +1,145 @@
+
+import pandas as pd
+from dash import Dash, html
+import dash_cytoscape as cyto
+cyto.load_extra_layouts()
+
+#run this app to run only the website
+app = Dash(__name__)
+
+
+
+#load colors for the nodes 
+def interpolate_color(color1, color2, factor):
+    """ Interpolate between two RGB colors """
+    r = int(color1[0] * (1 - factor) + color2[0] * factor)
+    g = int(color1[1] * (1 - factor) + color2[1] * factor)
+    b = int(color1[2] * (1 - factor) + color2[2] * factor)
+    return (r, g, b)
+
+def number_to_rgb_gradient(number, min_val, max_val,csv):
+    """ Convert number to RGB gradient """
+    normalized_value = (number - min_val) / (max_val - min_val)
+    if csv==1:
+        color1 = (255,255,204)  # Lightest color (white)
+        color2 = (255,0,0)  # Brightest color (red)
+        
+    if csv==2:
+        color1 = (240,248,255) # Lightest color (blue)
+        color2 = (0, 2, 29)  # Brightest color (dark blue)    
+    return interpolate_color(color1, color2, normalized_value)
+
+
+    
+def visualization(csv,attribute_csv,attribute_csv2):
+    #drop the duplicates and edges to themselves
+    Edge_df=csv.drop_duplicates(keep="first").reset_index(drop=True)
+    Edge_df = Edge_df[Edge_df['from'] != Edge_df['to']].reset_index(drop=True)
+    
+    #remove nodes without edges and insert the nodes and edges into a dataset readable by the network generator
+    attribute_csv_empty = attribute_csv[attribute_csv['Node'].isin(Edge_df['from']) | attribute_csv['Node'].isin(Edge_df['to'])].reset_index()
+    for i in range(len(attribute_csv_empty)):
+        attribute_csv_empty.loc[i, 'Count'] = len(Edge_df[Edge_df.to == attribute_csv_empty.loc[i,'Node']])
+        attribute_csv_empty.loc[i, 'Count'] = attribute_csv_empty.loc[i, 'Count']+len(Edge_df[Edge_df['from'] == attribute_csv_empty.loc[i,'Node']])
+
+    min_val = min(attribute_csv_empty['Count'])
+    max_val = max(attribute_csv_empty['Count'])
+        
+    # Convert each number to RGB gradient
+    rgb_gradients = [number_to_rgb_gradient(num, min_val, max_val,1) for num in attribute_csv_empty['Count']]
+    attribute_csv_empty['gradient']=rgb_gradients
+    #adds the nodes
+    nodes = [{
+                "data": {"id": str(attribute_csv_empty.loc[i, "Node"]), 
+                       "label": str(attribute_csv_empty.loc[i, "Node"]),
+                       'color': 'rgb'+str(attribute_csv_empty.loc[i, "gradient"])
+                       }
+                } for i in range(len(attribute_csv_empty))]
+    
+    
+    #copy paste because it's I'm lazy

```

Comment:
lolll hahah, prima maar mss even een comment zetten van copy paste (to be cleaned later) oid, zodat als we dit aan apollo geven straks we niet per ongeluk grappige comments erin laten staan

Reply:
thanks :)